engine/src/main/scala/org/scalarules/dsl/nl/grammar/DslLoop.scala

```diff
@@ -0,0 +1,32 @@
+package org.scalarules.dsl.nl.grammar
+
+import org.scalarules.dsl.nl.grammar.DslCondition.factFilledCondition
+import org.scalarules.engine._
+
+trait DslLoopWordTrait {
+  val Loop = new DslLoopWord
+}
+class DslLoopWord {
+  def over[A](listFact: ListFact[A]): DslLoopOverWord[A] = new DslLoopOverWord(listFact)
+}
+
+class DslLoopOverWord[A](listFact: ListFact[A]) {
+  def per(fact: SingularFact[A]): DslLoop[A] = new DslLoop(listFact, fact)
+}
+
+class DslLoop[A](listFact: ListFact[A], fact: SingularFact[A]) {
+  def doe[B, C](dslEvaluation: SubRunData[B, C]): SubRunData[List[B], A] = {
+    val resultFact: Fact[List[B]] = new ListFact("LoopFact")
```

Comment:
How is this going to hold up when we are nesting loops? I remember trying this just before `SubBerekening` was introduced and not having a solution for this.